azurelinuxagent/common/event.py

```diff
@@ -242,57 +242,55 @@ def is_period_elapsed(self, delta, h):
+                           event_time_opcode_name=event_time_opcode_name, event_pid=event_pid, event_tid=event_tid,
+                           task_name=task_name, keyword_name=keyword_name)
             self.periodic_events[h] = datetime.now()
 
-    def add_event(self,
-                  name,
-                  op=WALAEventOperation.Unknown,
-                  is_success=True,
-                  duration=0,
-                  version=CURRENT_VERSION,
-                  message="",
-                  evt_type="",
-                  is_internal=False,
-                  log_event=True):
+    def add_event(self, name, op=WALAEventOperation.Unknown, is_success=True, duration=0, version=str(CURRENT_VERSION),
+                  message="", evt_type="", is_internal=False, log_event=True,
+                  event_time_opcode_name=None, event_pid=0, event_tid=0, task_name="",
+                  keyword_name=""):
 
         if (not is_success) and log_event:
             _log_event(name, op, message, duration, is_success=is_success)
 
-        self._add_event(duration, evt_type, is_internal, is_success, message, name, op, version, event_id=1)
+        self._add_event(duration, evt_type, is_internal, is_success, message, name, op, version, event_id=1,
+                        event_time_opcode_name=event_time_opcode_name, event_pid=event_pid, event_tid=event_tid,
+                        task_name=task_name, keyword_name=keyword_name)
 
-    def _add_event(self, duration, evt_type, is_internal, is_success, message, name, op, version, event_id):
+    def _add_event(self, duration, evt_type, is_internal, is_success, message, name, op, version, event_id,
+                   event_time_opcode_name=None, event_pid=0, event_tid=0, task_name=None, keyword_name=None):
         event = TelemetryEvent(event_id, TELEMETRY_EVENT_PROVIDER_ID)
         event.parameters.append(TelemetryEventParam('Name', name))
         event.parameters.append(TelemetryEventParam('Version', str(version)))
         event.parameters.append(TelemetryEventParam('IsInternal', is_internal))
         event.parameters.append(TelemetryEventParam('Operation', op))
-        event.parameters.append(TelemetryEventParam('OperationSuccess',
-                                                    is_success))
+        event.parameters.append(TelemetryEventParam('OperationSuccess', is_success))
         event.parameters.append(TelemetryEventParam('Message', message))
         event.parameters.append(TelemetryEventParam('Duration', duration))
         event.parameters.append(TelemetryEventParam('ExtensionType', evt_type))
-        event.parameters.append(TelemetryEventParam('ContainerId',
-                                                    os.environ.get(CONTAINER_ID_ENV_VARIABLE, "UNINITIALIZED")))
 
+        self._add_default_parameters_to_event(event, container_id=os.environ.get(
+            CONTAINER_ID_ENV_VARIABLE, "UNINITIALIZED"), event_time_opcode_name=event_time_opcode_name,
```

Comment:
(i moved CONTAINER_ID_ENV_VARIABLE to event.py)

Reply:
let's make 

> os.environ.get(CONTAINER_ID_ENV_VARIABLE, "UNINITIALIZED")

a helper function to ensure consistency

Reply:
i'll rebase against develop, and then add a new helper.